File: app/src/main/java/com/moby/app/UserDao.java
package com.moby.app;

import androidx.room.*;
import java.util.List;

/**
 * UserDao — Data Access Object untuk Entity User.
 * TODO: Tambah query sesuai kebutuhan.
 */
@Dao
public interface UserDao {

    @Insert
    void insert(User user);

    @Query("SELECT * FROM users")
    List<User> getAll();

    @Delete
    void delete(User user);
}

File: app/src/main/java/com/tes/aoo/UserDao.kt
package com.tes.aoo

import androidx.room.*

/**
 * UserDao — Data Access Object untuk User.
 * TODO: Tambah query sesuai kebutuhan
 */
@Dao
interface UserDao {

    @Insert
    suspend fun insert(user: User)

    @Query("SELECT * FROM users ORDER BY id ASC")
    fun getAll(): List<User>

    @Query("SELECT * FROM users WHERE id = :id LIMIT 1")
    fun findById(id: Int): User?

    @Update
    suspend fun update(user: User)

    @Delete
    suspend fun delete(user: User)

    @Query("DELETE FROM users")
    suspend fun deleteAll()
}
